qiskit/algorithms/optimizers/p_bfgs.py

```diff
@@ -135,17 +158,19 @@ def optimize(
             )
 
         queue = multiprocessing.Queue()
+
+        # TODO: are automatic bounds a good idea? What if the circuit parameters are not
+        # just from plain Pauli rotations but have a coefficient?
```

Comment:
In general in the past, when used via VQE, bounds were set as informed via the ansatz, With VQE more taking QuantumCircuits that may not have this information like our pre-formed ansatzes do one wonders if bounds ought to be optionally settable via VQE, like one can do with the initial point. Maybe these default bounds may not always be applicable and/or work well, so a way to supply bounds at least this would give some way to override things. One could also change this optimizer to make bounds required instead of supported - that way bounds would have to be supplied.

Reply:
This is a parallel implementation that used multiple BFGS instances - it chose several different initial points within the bounds and would select the best result. I was commenting around the TODO that was added - but yes looking at addressing bounds ought to be some separate PR - see also this issue #5713 from some back